src/js/http_outgoing.js

```diff
@@ -152,8 +152,8 @@ OutgoingMessage.prototype._storeHeader = function(statusLine) {
 
 OutgoingMessage.prototype.setHeader = function(name, value) {
 
-  if (this._headers === null) {
-    this._headers = {};
+  if (util.isNull(this._headers)) {
+    this._headers = [];
```

Comment:
First of all, this commit is to improve code coverage. So please don't fix any original code. Plus, I don't agree this change. Kindly let me know why using `Array` is needed. Seeing the code below, `this._headers` is still used as `Object` although you set it as `Array`.